1-Draft/RFCNNNN-ML-Based-Command-Line-Completion.md

```diff
@@ -0,0 +1,136 @@
+---
+RFC: RFC
+Author: Paul Higinbotham, Travis Plunk, Dongbo Wang
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Engine
+Comments Due: March 30, 2018
+---
+
+# Machine Learning based Command Line Completion Suggestions for PowerShell
+
+The RFC proposes enhancing PowerShell's existing command line completion with machine learning based suggestions. 
+
+## Motivation
+
+PowerShell's existing command line completion looks for closest matches of known commands, cmdlets, and cmdlet parameters. The match is performed on the text fragment currently typed on the command line. Commands are shown alphabetically and parameters based on parameter set. 

```

Comment:
Please use semantic line breaks so that future changes are more easily readable (makes it so you can see a line change vs an paragraph change)

Reply:
Will do.